src/storage/compactor/src/server.rs

```diff
@@ -134,15 +134,14 @@ pub async fn prepare_start_parameters(
     );
 
     let memory_limiter = Arc::new(MemoryLimiter::new(compactor_memory_limit_bytes));
-    let storage_memory_config = extract_storage_memory_config(&config);
     let memory_collector = Arc::new(HummockMemoryCollector::new(
         sstable_store.clone(),
         memory_limiter.clone(),
         storage_memory_config,
     ));
 
     let heap_profiler = HeapProfiler::new(
-        system_memory_available_bytes(),
+        compactor_opts.compactor_total_memory_bytes,
```

Comment:
Is this correct? The [documentation](https://github.com/risingwavelabs/risingwave/blob/4f5f97a65656b6541c62e0d7d396801cb1dc4cf7/src/common/heap_profiling/src/profiler.rs#L41) of HeapProfiler says:

> `total_memory` must be the total available memory for the process.
> It will be compared with the process resident memory.

Reply:
Correct me if I'm wrong.

I refer to cn's approach, where compactor_opts.compactor_total_memory_bytes represents the memory limit for the current process, whether it is a dedicated or standalone deployment.

I understand that after https://github.com/risingwavelabs/risingwave/pull/19477, standalone will also set total_memory_bytes for cn, so the current pr is similar to CN.